apps/fares/lib/transfer.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Fares.Transfer do
+  @moduledoc """
+    Tools for handling logic around transfers between transit legs and modes.
+    The MBTA allows transfers between services depending on the fare media used
+    and the amount paid.
+
+    This logic may be superseded by the upcoming fares work.
+  """
+  alias Routes.{Repo, Route}
+  alias TripPlan.{Leg, NamedPosition, TransitDetail}
+
+  @type fare_atom :: Route.gtfs_route_type() | :inner_express_bus | :outer_express_bus
+
+  # Paying a single-ride fare for the first may get you a transfer to the second
+  # (can't be certain, as it depends on media used)!
+  @single_ride_transfers %{
+    :bus => [:subway, :bus],
+    :subway => [:subway, :bus],
+    :inner_express_bus => [:subway, :bus],
+    :outer_express_bus => [:subway, :bus]
+  }
+
+  @doc "Searches a list of legs for evidence of an in-station subway transfer."
+  @spec is_subway_transfer?([Leg.id_t()]) :: boolean
```

Comment:
Here's one thing that dialyzer is complaining about: `Leg` doesn't have an `id_t` type.